frontend/src/stimulus/controllers/dynamic/work-packages/relations-tab/scroll.controller.ts

```diff
@@ -0,0 +1,17 @@
+import { Controller } from '@hotwired/stimulus';
+
+export default class ScrollController extends Controller {
+  connect() {
+    this.waitForRenderAndAct();
+  }
+
+  waitForRenderAndAct() {
+    const element = document.querySelector('[data-scroll-to="true"]');
+
+    if (element) {
+      element.scrollIntoView({ behavior: 'smooth', block: 'center' });
+    } else {
+      requestAnimationFrame(this.waitForRenderAndAct.bind(this));
```

Comment:
This creates like an infinte loop when there is no scroll target (e.g. on a fresh page reload). The code is just executed endlessly.
Further, `requestAnimationFrame` is just meant for animations and will execute the code 60 times per second (assuming 60hertz screen), because that makes sense for animations. But you don't have an animation here.

Reply:
Hi @cbliard , @HDinger 
@ulferts and I investigated this issue more. The problem arises because a new row is added to the DOM, but it takes additional time to fully render and become visible in the view. As a result, we can't scroll to it immediately and need to use a timeout to give the Turbo.renderStream method enough time to render the new row.
Jonas encountered a similar issue in the activities tab: https://github.com/opf/openproject/blob/dev/frontend/src/stimulus/controllers/dynamic/work-packages/activities-tab/index.controller.ts#L213-L224